test/unit/BCoWPool/BCoWPool.tree

```diff
@@ -0,0 +1,8 @@
+BCoWPool::Finalize
```

Comment:
shouldn't we be testing `_afterFinalize` and check on BPool `finalize` that it calls it?

Reply:
i don't exactly like the `when preconditions are met` since there are no preconditions, i'd use `when called` for such scenarios